javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -270,19 +272,27 @@ object Integer {
   @inline def signum(i: scala.Int): scala.Int =
     if (i == 0) 0 else if (i < 0) -1 else 1
 
-  // Intrinsic
-  def numberOfLeadingZeros(i: scala.Int): scala.Int = {
-    // See Hacker's Delight, Section 5-3
-    var x = i
-    if (x == 0) {
-      32
+  @inline def numberOfLeadingZeros(i: scala.Int): scala.Int = {
+    if (linkingInfo.esVersion >= ESVersion.ES2015) js.Math.clz32(i)
+    else clz32Dynamic(i)
+  }
+
+  private def clz32Dynamic(i: scala.Int) = {
+    if (!Utils.isUndefined(js.Dynamic.global.Math.clz32)) {
```

Comment:
`Utils` is not in the minilib, so we can't use it here. We have to use `js.typeOf`. Otherwise LGTM.